### PR TITLE
feat(dp-token): allow validator to define keys not scoped to a mesh

### PR DESCRIPTION
**Pull request by nicoche**
This permits dpServer.authn.dpPRoxy.dpToken.validator.publicKeys to omit the .mesh field. Keys without the .mesh field will be considered to verify the signature of dataplane tokens, no matter the mesh of the dataplane.

### Checklist prior to review

- [x] Link to relevant issue https://kuma-mesh.slack.com/archives/CN2GN4HE1/p1698324155036599  --
- [x] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [x] Tests (Unit test, E2E tests, manual test on universal and k8s) --
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label) --

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

**Comment by jakubdyszkiewicz** [comment]
Nice!

Could you also add E2E test for it?
We could improve test/e2e_env/universal/auth/offline_auth.go:13
* generate another sample key here test/keys/README.md
* add a new entry in cpConfig#dpServer.authn.dpProxy.validator.publicKeys in test
* add another `It(` (or change to DescribeTable if you want) with the new key

This way we can validate this functionality and also check if we can combine meshed and not meshed keys.

Let me know if you need help

**Comment by nicoche** [comment]
Great! Honestly, I did not know where to look for, regarding the tests. I'll add that

**Comment by nicoche** [comment]
Should be good now, the e2e tests pass locally 🙂

File: test/keys/README.md [hunk 1]
```diff
@@ -1,4 +1,7 @@
 Keys for tests generated by executing:
 
-kumactl generate signing-key --format=pem > samplekey.pem
-kumactl generate public-key --signing-key-path=samplekey.pem > publickey.pem
+kumactl generate signing-key --format=pem > sample-key-static-1.pem
+kumactl generate public-key --signing-key-path=sample-key-static-1.pem > sample-publickey-static-1.pem
```
Comment (jakubdyszkiewicz) [inline test/keys/README.md:4]: I believe this key is used by unit tests as well so renaming will blow those tests up.

btw. key itself has no motion of the mesh. It's also used in context of other signing keys. I'd just name this
```
samplekey.pem
samplekey-2.pem
```
or if you want to rename it
```
samplekey-1.pem
samplekey-2.pem
```

Comment (nicoche) [reply test/keys/README.md]: Fair enough 🙂. I went with your first suggestion

**Review by jakubdyszkiewicz — APPROVED** [review]

**Pull request merged**